pkg/providers/ovirt/validation/vm-import-validation.go

```diff
@@ -159,22 +160,27 @@ func (validator *VirtualMachineImportValidator) processMappingValidationFailures
 	for _, failure := range failures {
 		message = withMessage(message, failure.Message)
 	}
+	instance := &v2vv1alpha1.VirtualMachineImport{}
+	err := validator.client.Get(context.TODO(), *vmiCrName, instance)
+	if err != nil {
+		return err
+	}
+	copy := instance.DeepCopy()
 
-	if len(failures) > 0 {
-		instance := &v2vv1alpha1.VirtualMachineImport{}
-		err := validator.client.Get(context.TODO(), *vmiCrName, instance)
-		if err != nil {
-			return err
-		}
-		copy := instance.DeepCopy()
+	invalid := len(failures) > 0
+	if invalid {
 		updateCondition(&copy.Status.Conditions, incompleteMappingRulesReason, message, false, v2vv1alpha1.Validating)
-		err = validator.client.Status().Update(context.TODO(), copy)
-		if err != nil {
-			return err
-		}
-		return fmt.Errorf("Mapping rules validation failed for %v. Reasons: %s", vmiCrName, message)
+	} else {
+		updateCondition(&copy.Status.Conditions, validationCompletedReason, "Validating completed successfully", true, v2vv1alpha1.Validating)
 	}
 
+	err = validator.client.Status().Update(context.TODO(), copy)
+	if err != nil {
+		return err
+	}
+	if invalid {
```

Comment:
Why not to check `invalid` one time and perform `updateCondition` and status update in specific if branch.

Reply:
That would duplicate the code responsible for the update call and its error handling in both branches.

Reply:
What about extracting it to a method?

Reply:
Having the update call in both branches will result in exactly the same number of lines as it is now:) The extracted method will only add to it.
I must admit, that when I look at the code changed as you suggested in the first comment, it does look more elegant, despite the duplication. I will change it this way.